libr/debug/p/native/bsd/bsd_debug.c

```diff
@@ -498,3 +498,39 @@ RList *bsd_desc_list(int pid) {
 	return ret;
 #endif
 }
+
+RList *bsd_thread_list(RDebug *dbg, int pid, RList *list) {
+#if __KFBSD__
+	int mib[4] = {CTL_KERN, KERN_PROC, KERN_PROC_PID | KERN_PROC_INC_THREAD, pid };
+	struct kinfo_proc *kp;
+	size_t len = 0;
+	size_t max;
+	int i = 0;
+
+	if (sysctl (mib, 4, NULL, &len, NULL, 0) == -1) {
+		r_list_free (list);
+		return NULL;
+	}
+
+	len += sizeof(*kp) + len / 10;
+	kp = malloc(len);
+	if (sysctl (mib, 4, kp, &len, NULL, 0) == -1) {
+		free (kp);
+		r_list_free (list);
+		return NULL;
+	}
+
+	max = len / sizeof(*kp);
+	for (i = 0; i < max; i ++) {
+		RDebugPid *pid_info;
+		pid_info = r_debug_pid_new (NULL, kp[i].ki_tid, kp[i].ki_uid, 's', 0);
```

Comment:
According to [kernel sources](https://github.com/freebsd/freebsd/blob/7e4664b0c30ad2ab36b196af2e02847cec202700/sys/sys/user.h#L175), it seems possible to obtain path from ki->ocomm(should also be available in comm [based on the implementation](https://github.com/freebsd/freebsd/blob/7e4664b0c30ad2ab36b196af2e02847cec202700/sys/kern/kern_proc.c#L738)) and status from ki->wchan instead of setting it to a default 's'. Would be nice to have those in `dpt` to stay consistent with the other debug implementations.

Reply:
true but ocomm is the not the full path.

Reply:
Better than an empty space, it will show the process' name for clarity.

Reply:
fair enough